File: src/main/java/komu/blunt/ast/BindGroup.java
package komu.blunt.ast;

import komu.blunt.types.checker.Assumptions;

import java.util.ArrayList;
import java.util.List;

public final class BindGroup {

    public final List<ExplicitBinding> explicitBindings = new ArrayList<>();
    public final List<List<ImplicitBinding>> implicitBindings = new ArrayList<>();
    
    public BindGroup(List<ExplicitBinding> explicitBindings,
                     List<ImplicitBinding> implicitBindings) {
        this.explicitBindings.addAll(explicitBindings);
        this.implicitBindings.add(implicitBindings);
    }

    public Assumptions assumptionFromExplicitBindings() {
        Assumptions.Builder builder = Assumptions.builder();
        
        for (ExplicitBinding b : explicitBindings)
            builder.add(b.name, b.scheme);

        return builder.build();
    }

}

File: src/main/kotlin/komu/blunt/ast/BindGroup.kt
package komu.blunt.ast

import java.util.ArrayList
import komu.blunt.types.checker.Assumptions

class BindGroup(explicitBindings: List<ExplicitBinding>, implicitBindings: List<ImplicitBinding>) {

    val explicitBindings: MutableList<ExplicitBinding> = ArrayList(explicitBindings)
    val implicitBindings: MutableList<List<ImplicitBinding>> = ArrayList(listOf(implicitBindings))

    fun assumptionFromExplicitBindings(): Assumptions {
        val builder = Assumptions.builder()

        for (binding in explicitBindings)
            builder[binding.name] = binding.scheme

        return builder.build()
    }
}
